pkg/odo/cli/describe/component.go

```diff
@@ -306,9 +309,30 @@ func (o *ComponentOptions) describeDevfileComponent(ctx context.Context) (result
 		// Display RunningOn field only if the feature is enabled
 		cmp.RunningOn = nil
 	}
+	updateWithRemoteSourceLocation(&cmp)
 	return cmp, devfileObj, err
 }
 
+func updateWithRemoteSourceLocation(cmp *api.Component) {
+	components, err := cmp.DevfileData.Devfile.GetComponents(common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{ComponentType: v1alpha2.ContainerComponentType},
+	})
+	if err != nil {
+		return
+	}
+	for _, comp := range components {
+		if *comp.Container.MountSources {
```

Comment:
Can `comp.Container.MountSources` be nil?
I saw a `GetMountSources()` method which is supposed to handle this case and also return the appropriate boolean if `dedicatedPod` is set.

```suggestion
		if comp.Container.GetMountSources() {
```